File: hackerrank/30-days-of-code/linked-list/linked-list.py
#!/usr/bin/env python

class Node:
    def __init__(self,data):
        self.data = data
        self.next = None 
class Solution: 
    def display(self,head):
        current = head
        while current:
            print(current.data,end=' ')
            current = current.next

    def setData(self, data):
        self.data = data

    def getData(self):
        return self.data

    def insert(self,head,data): 
        newNode = Node()
        newNode.setData(data)

        current = self.head
        while current.getNext() != None:
            current = current.getNext()
        current.setNext(newNode)
        self.length += 1

mylist= Solution()
T=int(input())
head=None
for i in range(T):
    data=int(input())
    head=mylist.insert(head,data)    
mylist.display(head); 	  

File: hackerrank/30-days-of-code/linked-list/LinkedList.java
import java.io.*;
import java.util.*;
class Node {
	int data;
	Node next;
	Node(int d) {
        data = d;
        next = null;
    }
}
class LinkedList {
    public static  Node insert(Node head,int data) {
		Node node = new Node(data);
    	Node start = head;
    	if(head == null) {
    		head = new Node(data);
            // System.out.println(head.data);
        } else {
    		while(start.next != null) {
    			start = start.next;
            }
            start.next= node;
        }
        return head;
    }
    public static void display(Node head) {
        Node start = head;
        while(start != null) {
            System.out.print(start.data + " ");
            start = start.next;
        }
        System.out.println();
    }
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        Node head = null;
        int N = sc.nextInt();
        while(N-- > 0) {
            int ele = sc.nextInt();
            head = insert(head,ele);
        }
        display(head);
        sc.close();
    }
}
